fix(dominikh/go-tools/staticcheck): fix versioning

takumin opened the pull request:
- GitHub release versioning changed
  - newer: https://github.com/dominikh/go-tools/releases/tag/v0.3.3
  - older: https://github.com/dominikh/go-tools/releases/tag/2022.1.2

takumin committed:
fix(dominikh/go-tools/staticcheck): fix versioning

- GitHub release versioning changed
  - https://github.com/dominikh/go-tools/releases/tag/v0.3.3

takumin committed:
fix(dominikh/go-tools/staticcheck): fix darwin install

suzuki-shunsuke commented on pkgs/dominikh/go-tools/staticcheck/registry.yaml:
```diff
@@ -4,19 +4,21 @@ packages:
         src: staticcheck/staticcheck
     supported_envs:
-      - darwin
+      - darwin/amd64
```
darwin/arm64 is supported by `rosetta2: true`.
Basically, we don't use `darwin/amd64`.

suzuki-shunsuke commented on pkgs/dominikh/go-tools/staticcheck/registry.yaml:
```diff
@@ -4,19 +4,21 @@ packages:
       algorithm: sha256
       pattern:
-        checksum: ^(\b[A-Fa-f0-9]{64}\b)
+        checksum: "^(\b[A-Fa-f0-9]{64}\b)"
```
```suggestion
        checksum: "^(\\b[A-Fa-f0-9]{64}\\b)"
```

If you use double quote `"`, you have to escape `\`.

suzuki-shunsuke commented on registry.yaml:
```diff
@@ -5001,21 +5001,23 @@ packages:
       algorithm: sha256
       pattern:
-        checksum: ^(\b[A-Fa-f0-9]{64}\b)
+        checksum: "^(\b[A-Fa-f0-9]{64}\b)"
```
```suggestion
        checksum: "^(\\b[A-Fa-f0-9]{64}\\b)"
```

suzuki-shunsuke committed:
Apply suggestions from code review

suzuki-shunsuke commented on pkgs/dominikh/go-tools/staticcheck/pkg.yaml:
```diff
@@ -1,3 +1,3 @@
-  - name: dominikh/go-tools/staticcheck
-    version: 2022.1
+  - name: dominikh/go-tools/staticcheck@v0.3.3
+  - name: dominikh/go-tools/staticcheck@2022.1.2
```
```suggestion
```

This isn't needed.

suzuki-shunsuke committed:
Apply suggestions from code review

suzuki-shunsuke commented on pkgs/dominikh/go-tools/staticcheck/registry.yaml:
```diff
@@ -4,19 +4,21 @@ packages:
     repo_name: go-tools
     description: Go static analysis, detecting bugs, performance issues, and much more
-    asset: staticcheck_{{.OS}}_{{.Arch}}.tar.gz
+    asset: staticcheck_{{.OS}}_{{.Arch}}.{{.Format}}
+    format: tar.gz
```
I don't think this update is needed.
If the archive format is different by OS, we have to parameterize the format, but in that case we don't have to parameterize the format.

takumin committed:
fix(dominikh/go-tools/staticcheck): reflect comments

- https://github.com/aquaproj/aqua-registry/pull/7088/files#r996279914
- https://github.com/aquaproj/aqua-registry/pull/7088/files#r996278093

suzuki-shunsuke commented:
* https://github.com/dominikh/go-tools/releases/tag/v0.3.3: The asset for darwin/arm64 wasn't released
* https://github.com/dominikh/go-tools/releases/tag/2022.1.2: [staticcheck_darwin_arm64.tar.gz](https://github.com/dominikh/go-tools/releases/download/2022.1.2/staticcheck_darwin_arm64.tar.gz) was released

suzuki-shunsuke commented:
https://github.com/aquaproj/aqua-registry/pull/7088#issuecomment-1279716013

Hmm. `2022.1.2` isn't semver, so we can't use `semVer` in `version_constraint`..

suzuki-shunsuke committed:
fix(dominikh/go-tools/staticcheck): set version_overrides

suzuki-shunsuke committed:
fix(dominikh/go-tools/staticcheck): fix supported_envs

aqua doesn't support platform other than windows, macOS, and Linux.
So in this case `amd64` is enough.

suzuki-shunsuke commented:
Released. https://github.com/aquaproj/aqua-registry/releases/tag/v3.76.0
